DataDog/dd-trace-rb
# Add telemetry environment flag

**What does this PR do?**

This PR adds an environment variable `DD_INSTRUMENTATION_TELEMETRY_ENABLED` that will be used to determine whether or not telemetry requests should be sent. By default, telemetry is enabled (as per the spec [here](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v1/how-to-use.md#enablement)).

**Motivation**

This flag will control whether or not telemetry data should be collected and sent.

**Additional Notes**

* I set the variable to be false when running tests (in the `test-macos.yml` file) to prevent it from attempting to collect+send telemetry data on startup - is there a better way of preventing telemetry from affecting the rest of the tests?

**How to test the change?**

Unit tests were added to verify that the configuration gets set properly based on the env variable
